src/core/exec-invoke.c

```diff
@@ -3417,7 +3417,7 @@ static int apply_mount_namespace(
                 .protect_kernel_tunables = needs_sandboxing && context->protect_kernel_tunables,
                 .protect_kernel_modules = needs_sandboxing && context->protect_kernel_modules,
                 .protect_kernel_logs = needs_sandboxing && context->protect_kernel_logs,
-                .protect_hostname = needs_sandboxing && context->protect_hostname,
+                .protect_hostname = needs_sandboxing && context->protect_hostname == PROTECT_HOSTNAME_YES,
```

Comment:
i think we should not use a different type here. i.e. please pass this down as ProtectHostname proper, even if the namespace code doesn#t really care about the distinction of two of the values. But it's really weird to use one type for this at some place and a different on another neeldessly.

hence:

```c
protect_hostname = needs_sandboxing ? context->protect_hostname : PROTECT_HOSTNAME_NO,
```

Reply:
-> #35580